src/chain/chaininterface.rs

```diff
@@ -24,6 +25,9 @@ pub trait ChainWatchInterface: Sync + Send {
 
 	fn register_listener(&self, listener: Weak<ChainListener>);
 	//TODO: unregister
+
+	/// Gets the chain currently watched
+	fn get_network(&self) -> Network;
```

Comment:
Actually while you're at it, why not just impelement the other TODO and make this
fn get_chain_txo(genesis_hash, txid, output_index) -> Result<(Script, Value), enum { IDontImplementThat, ItsNotThere, DontKnowThatChain })>

Reply:
I'm on it